docs: update server streaming mode documentation (#9519)
ggml-org/llama.cpp: examples/server/README.md

node index.js
```

Alternative script to test streaming mode (chunk splitting and error handling should be enhanced for production use):

```javascript
(async () => {
    const response = await fetch("http://localhost:8080/completion", {
        method: "POST",
        body: JSON.stringify({
            prompt: "To write an essay quickly",
            n_predict: 256,
            stream: true
        })
    })
    for await (const chunk of response.body.pipeThrough(new TextDecoderStream("utf-8"))) {

>         for (const event of chunk.split(/(?<=\n\n)/v)) {

Reviewer: This solution is OK for being a POC, but in reality, you may want to use `readline` because reverse proxy can cut the data chunk mod-way

cd llama-client
```

Create an index.js file and put this inside:

```javascript

> const prompt = `Building a website can be done in 10 simple steps:`

Reviewer: ```suggestion
const prompt = "Building a website can be done in 10 simple steps:"
```

**Response format**

> - Note: In streaming mode (`stream`), only `content` and `stop` will be returned until end of completion. Responses currently use a format resembling the [Server-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events) standard, but the `EventSource` interface cannot be used due to its lack of `POST` request support.

Reviewer: ```suggestion
- Note: In streaming mode (`stream`), only `content` and `stop` will be returned until end of completion. Responses are sent using the [Server-sent events](https://html.spec.whatwg.org/multipage/server-sent-events.html) standard. Note: the browser's `EventSource` interface cannot be used due to its lack of `POST` request support.
```

Removing the word "resembling" because the SSE standard doesn't specify if it supports GET or POST request. So we are already compliant with the standard. It's the browser that only implement GET for `EventSource`
